### ISSUE TITLE
Created a Bluetooth script for the FNB58

### ISSUE THREAD

**Issue by parkerlreed** [baryluk/fnirsi-usb-power-data-logger#25]
Took some time this morning to sniff out the Android application with btsnoop/wireshark.

Figured out enough the notification packet to get the main three values of voltage, amperage, and wattage

https://gist.github.com/parkerlreed/0ce45e907ce536a0541afb90b5b49350

Hope this helps.

**Comment by baryluk** [comment]
That is pretty cool. Good job.

I do not have any Bluetooth host / dongle to use with Linux, and I always have issues with Bluetooth anyway. I think my meters do have BT support, but i never tested it, even with Android.

From the look at your code, the data format is different, and there are other things that are different of course. But it looks simple and straightforward.

I will test it one day. Maybe it is not that a bad idea, to reduce number of cables.

Do you know what is the frequency of sample updates when using Bluetooth?

**Comment by parkerlreed** [comment]
Looks to be the same 4 lines per second 

```
07:06:19 - 4 lines per second
```

**Comment by chris4git** [comment]
> Took some time this morning to sniff out the Android application with btsnoop/wireshark.
> 
> Figured out enough the notification packet to get the main three values of voltage, amperage, and wattage
> 
> https://gist.github.com/parkerlreed/0ce45e907ce536a0541afb90b5b49350
> 
> Hope this helps.

Hi,
I try to use your script with my FNB58, unfortunately it receives nothing 

```
python3 ./fnirsi-fnb58.py 
Connected to Bluetooth device.
Wrote value: aa8100f4
Wrote value: aa8200a7
Notifications enabled.
```
parse_data() is not called.
Any hint ?

Chris

**Comment by chris4git** [comment]
> Hi, I try to use your script with my FNB58, unfortunately it receives nothing
> 
> ```
> python3 ./fnirsi-fnb58.py 
> Connected to Bluetooth device.
> Wrote value: aa8100f4
> Wrote value: aa8200a7
> Notifications enabled.
> ```
> 
> parse_data() is not called. Any hint ?
> 
> Chris

It seems related to the USB port in use.
Without USB connection your script works 👍 
Thanks for sharing.

Chris

**Comment by parkerlreed** [comment]
Yeah it can't USB and Bluetooth at the same time. Glad you figured it out.
